xline/src/server/command.rs

```diff
@@ -598,6 +599,60 @@ impl CurpCommand for Command {
     }
 }
 
+/// Generate `Command` proposal from `Request`
+pub(super) fn command_from_request_wrapper<S>(
+    propose_id: ProposeId,
+    wrapper: RequestWithToken,
+    lease_storage: Option<&LeaseStore<S>>,
+) -> Command
+where
+    S: StorageApi,
+{
+    #[allow(clippy::wildcard_enum_match_arm)]
+    let keys = match wrapper.request {
+        RequestWrapper::RangeRequest(ref req) => {
+            vec![KeyRange::new(req.key.as_slice(), req.range_end.as_slice())]
+        }
+        RequestWrapper::PutRequest(ref req) => vec![KeyRange::new_one_key(req.key.as_slice())],
+        RequestWrapper::DeleteRangeRequest(ref req) => {
+            vec![KeyRange::new(req.key.as_slice(), req.range_end.as_slice())]
+        }
+        RequestWrapper::TxnRequest(ref req) => req
+            .compare
+            .iter()
+            .map(|cmp| KeyRange::new(cmp.key.as_slice(), cmp.range_end.as_slice()))
+            .collect(),
+        RequestWrapper::LeaseRevokeRequest(ref req) => {
+            let Some(lease_storage) = lease_storage else {
+                panic!("lease_storage is None when creating command of LeaseRevokeRequest")
```

Comment:
The panic message should focus more on the expectation rather than the fact. "is None" is a fact while "should be Some(_)" is this method's expectation. I prefer using expectation in the panic message.

Reply:
Fixed